gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/resolver/GfxdPartitionByExpressionResolver.java

```diff
@@ -670,6 +686,9 @@ public boolean okForColocation(GfxdPartitionResolver rslvr) {
       // Minimal check do determine if both have them have expr node at least
       if (this.exprCompiler != null && passed.exprCompiler == null
           || this.exprCompiler == null && passed.exprCompiler != null) {
+        if(snappyStore){ //escaping the check for snappystore.
+          return true;
+        }
         return false;
```

Comment:
Sumedh, I agree with you on not using expression based partitioner . I was exploring and it looked like the easiest way to implement this feature. But we can not go ahead with the de serialization overhead every time we compute hashcode(). This will defeat the purpose. I am now trying to use the column based partitioner. 

Two code paths we need to check. One is from computeHashCode() , which we need to add another method in DataValueDescriptor to give a hashcode which is similar to Spark's hashcode. This part is easy. Another is when we compute hashcode from row byte[]. Couple of DataTypes like SQLDate needs to change to serialize extra information that we need make it similar to java.util.Date() hashcode. 
SQLChar also needs some modification.